libsrc/grabber/v4l2/V4L2Grabber.cpp

```diff
@@ -1013,6 +1087,10 @@ int V4L2Grabber::read_frame()
 
 bool V4L2Grabber::process_image(const void *p, int size)
 {
+
+	if ( (_fpsSoftwareDecimation > 1) && ((_currentFrame++) % _fpsSoftwareDecimation == 0))
```

Comment:
Why do we need this when we can change capture frequency ?

Reply:
FPS settings from the device configuration is insufficient for high quality 1080/4k grabbers due to lack of resources on some configuration (Rpi3 in my case).

As for ezcap 269 https://postimg.cc/F1sgFK3N there is only a choice between 30 and 60fps: nothing between, nothing less. That causes some delay between movie on the TV and colors on the leds. Frame decimation somehow reduces that problem but works bad with mjpeg encoding in lower resolution (jpeg artifact have quite impact). Rejecting certain frames before processing them allows us to use higher resolutions or abandon frame decimation or to have less impact on CPU. All that reduces delay TV<->LED if that occurs for a price of lower FPS for Hyperion.

PS: USB 2.0 with mjpeg  bus is not a problem even on Rpi3 https://postimg.cc/4Yh1CFtc with 30 or 60 fps. But the CPU usage are over 100% mostly for only one core. Only 640x480 dec =1 and 800x600 dec =2 are acceptable with 30 FPS. Reducing FPS and CPU usage solved the lag problem in my case.

Reply:
I didn't know that frame rate setting is restricted to predefined values in some devices. 
```

                // Check if the device is able to accept a capture framerate set.
                if (streamparms.parm.capture.capability == V4L2_CAP_TIMEPERFRAME)
                {
                        streamparms.parm.capture.timeperframe.numerator = 1;
                        streamparms.parm.capture.timeperframe.denominator = _fps;
                        (-1 == xioctl(VIDIOC_S_PARM, &streamparms))
                        ?       Debug(_log, "Frame rate settings not supported.")
                        :       Debug(_log, "Set framerate to %d fps", streamparms.parm.capture.timeperframe.denominator);
                }
```
Does this also apply to resolution settings ? I mean, isn't it possible to reduce the resolution to something lower than 640x480 ? Because those values are higher than what we provide as default in other grabbers (Picture decimation is 8 by default, so a FHD image becomes 240x135). 

Or you tried and didn't like the results with lower resolution ? If so, what difference did it make to use a higher resolution ?

Reply:
I have attached screenshots:
800x600 30fps decimation = 1 => CPU-core reaches it's limit
![perf800600_30](https://user-images.githubusercontent.com/69086569/89455161-25d98280-d762-11ea-9d26-d179a1a8610e.jpg)

800x600 custom 5fps decimation = 1 => no change CPU-core reaches it's limit, hardware FPS limit override it
![perf800600_3](https://user-images.githubusercontent.com/69086569/89455219-3f7aca00-d762-11ea-854d-ef516e73a74f.jpg)

and 800x600 new software frame skipping(5 for 30FPS) so 6fps, decimation = 1 => CPU-core usage lowered to 60%
![perf800600_skipping6](https://user-images.githubusercontent.com/69086569/89455317-6a651e00-d762-11ea-9c31-3d5098ef041a.jpg)

But I've discovered that when I have added that option to the global config (for my use I hardcoded it) I made a school mistake in modulo formula for frame skipping :anguished: : 
IN PULL REQUEST:
`if ( (_fpsSoftwareDecimation > 1) && ((_currentFrame++) % _fpsSoftwareDecimation == 0))
+		return false;`
SHOULD BE OF COURSE:
`if ( (_fpsSoftwareDecimation > 1) && ((_currentFrame++) % _fpsSoftwareDecimation **!=** 0))
+		return false;`
otherwise it does skipping but not I mean to.